resourcemanager/recaser/recase.go

```diff
@@ -81,9 +138,13 @@ func fixSegment(input, segment string) string {
 // so it can be used as a key to extract the correct id from knownResourceIds
 func buildInputKey(input string) (*string, bool) {
 
-	// don't attempt to build a key if this isn't a standard resource id
+	// Attempt to determine if this is just missing a leading slash and prepend it if it seems to be
 	if !strings.HasPrefix(input, "/") {
```

Comment:
Understood. We have seen cases of Azure APIs returning resource IDs with the leading slash missing, and it's low effort for us to try it rather than having users need to concat it on input themselves. As you say, if we don't get a key-match, we don't get an error either.